packages/datagateway-search/src/card/datasetSearchCardView.component.tsx

```diff
@@ -59,79 +66,165 @@ const DatasetCardView = (props: DatasetCardViewProps): React.ReactElement => {
+      {
+        searchText,
+        startDate,
+        endDate,
+        sort,
+        minCount: minNumResults,
+        maxCount: maxNumResults,
+        restrict,
+        facets: [{ target: 'Dataset' }],
+      },
+      filters
+    );
+
+  const {
+    selectedFacetFilters,
+    addFacetFilter,
+    removeFacetFilter,
+    applyFacetFilters,
+    haveUnappliedFilters,
+  } = useFacetFilters();
+
+  useSearchResultCounter({
+    isFetching,
+    dataSearchType: 'Dataset',
+    searchResponses: data?.pages,
+    hasMore: hasNextPage,
   });
 
-  const [t] = useTranslation();
-
-  const { filters, sort, page, results } = React.useMemo(
-    () => parseSearchToQuery(location.search),
-    [location.search]
+  function mapSource(response: SearchResponse): SearchResultSource[] {
+    return response.results?.map((result) => result.source) ?? [];
+  }
+
+  function mapIds(response: SearchResponse): number[] {
+    return response.results?.map((result) => result.id) ?? [];
+  }
+
+  const mapFacets = React.useCallback(
+    (responses: SearchResponse[]): CVCustomFilters[] => {
+      const filters: { [dimension: string]: { [label: string]: number } } = {};
+      responses.forEach((response) => {
+        if (response.dimensions !== undefined) {
+          Object.entries(response.dimensions).forEach((dimension) => {
+            const dimensionKey = dimension[0];
+            const dimensionValue = dimension[1];
+            if (!Object.keys(filters).includes(dimensionKey)) {
+              filters[dimensionKey] = {};
+            }
+            Object.entries(dimensionValue).forEach((labelValue) => {
+              const label = labelValue[0];
+              const count =
+                typeof labelValue[1] === 'number'
+                  ? labelValue[1]
+                  : labelValue[1].count;
+              if (Object.keys(filters[dimensionKey]).includes(label)) {
+                filters[dimensionKey][label] += count;
+              } else {
+                filters[dimensionKey][label] = count;
+              }
+            });
+          });
+        }
+      });
+      return Object.entries(filters).map((dimension) => {
+        const dimensionKey = dimension[0].toLocaleLowerCase();
+        const dimensionValue = dimension[1];
+        return {
+          label: t(dimensionKey),
+          dataKey: dimensionKey,
+          dataKeySearch: dimensionKey.replace('dataset.', ''),
+          filterItems: Object.entries(dimensionValue).map((labelValue) => ({
+            name: labelValue[0],
+            count: labelValue[1].toString(),
+          })),
+          prefixLabel: true,
+        };
+      });
+    },
+    [t]
   );
 
-  const textFilter = useTextFilter(filters);
-  const dateFilter = useDateFilter(filters);
+  const parsedFilters = React.useMemo(() => {
+    const parsedFilters = {} as FiltersType;
+    Object.entries(filters).forEach((v) => {
+      parsedFilters[v[0].substring(8)] = v[1]; // "dataset." is 8 characters
```

Comment:
Rather than use `.substring(8)` here, is it perhaps clearer to use `.replace("dataset.", "")`? I think those will end up equivalent results (presuming the parsed filter always starts with `dataset.`)

Reply:
I don't think this code is needed anymore, as it is only used for showing the selected filters in `cardView`. Since it is already handled externally, `parsedFilters` can be safely removed.